x-pack/plugin/src/test/resources/rest-api-spec/test/api_key/10_basic.yml

```diff
@@ -214,10 +214,58 @@ teardown:
       security.invalidate_api_key:
         body:  >
             {
-              "id": "${api_key_id}"
+              "id": "${api_key_id_1}"
             }
   - length: { "invalidated_api_keys" : 1 }
-  - match: { "invalidated_api_keys.0" : "${api_key_id}" }
+  - match: { "invalidated_api_keys.0" : "${api_key_id_1}" }
+  - length: { "previously_invalidated_api_keys" : 0 }
+  - match: { "error_count" : 0 }
+
+  - do:
+      headers:
+        Authorization: "Basic YXBpX2tleV91c2VyOngtcGFjay10ZXN0LXBhc3N3b3Jk" # api_key_user
+      security.create_api_key:
+        body:  >
+          {
+            "name": "my-api-key-2",
+            "expiration": "1d",
+            "role_descriptors": {
+            }
+          }
+  - match: { name: "my-api-key-2" }
+  - is_true: id
+  - is_true: api_key
+  - is_true: expiration
+  - set: { id: api_key_id_2 }
+
+  - do:
+      headers:
+        Authorization: "Basic YXBpX2tleV91c2VyOngtcGFjay10ZXN0LXBhc3N3b3Jk" # api_key_user
+      security.create_api_key:
+        body:  >
+          {
+            "name": "my-api-key-3",
+            "expiration": "1d",
+            "role_descriptors": {
+            }
+          }
+  - match: { name: "my-api-key-3" }
+  - is_true: id
+  - is_true: api_key
+  - is_true: expiration
+  - set: { id: api_key_id_3 }
+
+  - do:
+      headers:
+        Authorization: "Basic YXBpX2tleV91c2VyOngtcGFjay10ZXN0LXBhc3N3b3Jk" # api_key_user
+      security.invalidate_api_key:
+        body:  >
+          {
+            "ids": [ "${api_key_id_2}", "${api_key_id_3}" ]
```

Comment:
Did you consider putting key `_1` in here, and checking `previously_invalidated_api_keys: 1` ?
It seems like a useful test of the error handling for multiple ids.

Reply:
Unfortunately no. The code filters out invalidated keys before calling `indexInvalidate`, i.e.
https://github.com/elastic/elasticsearch/blob/e3dd0a802f5b2fe1cff71addc080dc450c549423/x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/ApiKeyService.java#L765

Not sure if it is intended?

Reply:
I mean I will add key 1 here, but the response won't change.